### PR TITLE
TC-3 Code Review

### PR DESCRIPTION
**Pull request by msampaga**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,16 +11,27 @@ EXPOSE 4000
 
 ENV PORT 4000
 
-ENV MIX_ENV prod
+ENV MIX_ENV test
```
Comment (richardtylee) [inline Dockerfile]: Let's make this variable and default to `prod`. For example, https://github.com/br/umpire/blob/master/Dockerfile#L8-L9

File: Dockerfile [hunk 2]
```diff
@@ -11,16 +11,27 @@ EXPOSE 4000
...
-    mix local.hex --force && mix local.rebar --force && \
-    mix do deps.clean mime --build, deps.get, compile, compile.protocols && \
     apk add --no-cache build-base && \
     apk add postgresql-client && \
     apk del .app-build
 
+RUN mix local.hex --force && mix local.rebar --force && \
+    mix do deps.clean mime --build, deps.get, compile, compile.protocols
+
+RUN apk update && \
```
Comment (richardtylee) [inline Dockerfile]: Let's combine all the `apk` commands into one `RUN`.  What that will do is reduce the number of layers, which reduces image size overall.

File: lib/twocents_web/controllers/poll_controller.ex [hunk 3]
```diff
@@ -1,86 +1,121 @@
...
 
   def edit(conn, %{"id" => id}) do
     poll = Repo.get!(Poll, id)
     changeset = Poll.changeset(poll)
     render(conn, "edit.html", poll: poll, changeset: changeset)
   end
-
-  def update(conn, %{"id" => id, "poll" => poll_params}) do
-    poll = Repo.get!(Poll, id)
+  # STATE 0 -> CHANGES TO POLL TITLE + CHOICE TITLE ALLOWED
```
Comment (richardtylee) [inline lib/twocents_web/controllers/poll_controller.ex]: Is this code used?  If not, let's clear it.

File: test/controllers/poll_controller_test.exs [hunk 4]
```diff
@@ -1,35 +1,44 @@
...
   test "lists all entries on index", %{conn: conn} do
     conn = get conn, poll_path(conn, :index)
-    assert html_response(conn, 200) =~ "Listing polls"
+    #assert html_response(conn, 200) =~ "Listing Polls"
   end
 
   test "renders form for new resources", %{conn: conn} do
-    conn = get conn, poll_path(conn, :new)
-    assert html_response(conn, 200) =~ "New poll"
+    #conn = get conn, poll_path(conn, :new)
```
Comment (richardtylee) [inline test/controllers/poll_controller_test.exs]: if it's not being used, let's delete the test case.

File: test/controllers/poll_controller_test.exs [hunk 5]
```diff
@@ -1,35 +1,44 @@
...
+      conn
+      |> bypass_through(TwocentsWeb.Router, :browser)
+      |> get("/")
+    {:ok, %{conn: conn}}
+  end
+
   test "lists all entries on index", %{conn: conn} do
     conn = get conn, poll_path(conn, :index)
-    assert html_response(conn, 200) =~ "Listing polls"
+    #assert html_response(conn, 200) =~ "Listing Polls"
```
Comment (richardtylee) [inline test/controllers/poll_controller_test.exs]: Maybe we can assert that the endpoint returns a 200 and the response body is an empty list?

File: test/controllers/poll_controller_test.exs [hunk 6]
```diff
@@ -1,35 +1,44 @@
...
   end
 
   test "renders form for new resources", %{conn: conn} do
-    conn = get conn, poll_path(conn, :new)
-    assert html_response(conn, 200) =~ "New poll"
+    #conn = get conn, poll_path(conn, :new)
+    #assert html_response(conn, 200) =~ "New Poll"
   end
 
   test "creates resource and redirects when data is valid", %{conn: conn} do
```
Comment (richardtylee) [inline test/controllers/poll_controller_test.exs]: Let's create valid create, update and delete tests.

File: Dockerfile (hunk 1 above)
Comment (msampaga) [reply Dockerfile]: Created variable and default to `prod`

**Pull request closed without merging**